widgets/common/src/filters/RuleRow.tsx

```diff
@@ -0,0 +1,31 @@
+import type { FunctionComponent } from 'react';
+import RuleAttributeDropdown from './RuleAttributeDropdown';
+import RuleOperatorDropdown from './RuleOperatorDropdown';
+import RuleInput from './RuleInput';
+import RuleRemoveButton from './RuleRemoveButton';
+import { FormField, FormGroup } from './FormWrapper';
+
+interface RuleRowProps {
+    id: number;
+    onRemove: (id: number) => void;
```

Comment:
The placement of the parameter looks odd to me :smile: Will this component ever need to call it with something else than the `props.id`? I doubt it, but still, by having the API designed this way, we make it possible :slightly_smiling_face: 

What we could do is not pass the `id` here and force the parent to handle it. This way the parent can be user that the `id` will always be the `id` that is passed in the props, not something else.

```suggestion
    onRemove: () => void;
```

(or reuse the `onClick` type from `RuleRemoveButton` (see comment on `AddRuleButton` to see how to use that using the `ComponentProps` type)

Reply:
Of course. Removed using `id` and added typings.